File: implementations/java/vanilla/src/main/java/org/cwi/waebric/parser/ast/site/DirName.java
package org.cwi.waebric.parser.ast.site;

import org.cwi.waebric.parser.ast.ISyntaxNode;

public class DirName implements ISyntaxNode {

	private Directory directory;

	public Directory getDirectory() {
		return directory;
	}

	public void setDirectory(Directory directory) {
		this.directory = directory;
	}

	public ISyntaxNode[] getChildren() {
		return new ISyntaxNode[] { directory };
	}
	
	@Override
	public String toString() {
		return directory.toString();
	}
	
	@Override
	public boolean equals(Object obj) {
		return directory.equals(obj);
	}

}

File: implementations/csharp/vanilla/parser/ast/Site/DirName.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Parser.Ast.Site
{
    /// <summary>
    /// Node which contains DirName
    /// </summary>
    public class DirName : ISyntaxNode
    {
        #region Private Members

        private Directory DirectoryInstance;

        #endregion

        #region Public Methods

        /// <summary>
        /// Set directory of DirName
        /// </summary>
        /// <param name="directory">Directory to set</param>
        public void SetDirectory(Directory directory)
        {
            DirectoryInstance = directory;
        }

        /// <summary>
        /// Get directory of DirName
        /// </summary>
        /// <returns>Directory</returns>
        public Directory GetDirectory()
        {
            return DirectoryInstance;
        }

        /// <summary>
        /// Returns directory as string
        /// </summary>
        /// <returns>Directory</returns>
        public String ToString()
        {
            return DirectoryInstance.ToString();
        }

        #endregion
    }
}
